example/package-c/BUILD.bazel

```diff
@@ -23,24 +23,18 @@ haskell_library(
 
 # example where we combine files in srcs with ones found by the directive
 #
-# gazelle_haskell_modules:srcs: more-src/
+# gazelle_haskell_modules:srcs: explicit-src/Autodetect/
 haskell_library(
     name = "package-c-explicit-srcs",
-    srcs = ["src/PackageC/Other/B.hs"],
+    srcs = ["explicit-src/Explicit/DepOnAuto.hs"],
```

Comment:
Maybe you mean't `DepOnExplicit.hs` here? I still think it would be easier to get the test right if a module in a test binary imports all the modules from the library that are supposed to be present.

Reply:
The module is the correct one, I think.

I have the following setup:
* `Root.hs` - autodetected, depends on nothing
* `DepOnAuto.hs` - present explicitly, depends on `Root`, to check if explicit srcs can depend on autodetected ones
* `DepOnExplicit.hs` - autodetected (and consequently also explicitly to test for what happens when a module is present in both places), depends on a module that's explicit, to test if autodetected modules can depend on explicit ones.

Is there something wrong with this?

Reply:
Root isn't autodetected after the latest modifications to `package-c-explicit-srcs`.

In any case, I'm finding brittle enough reasoning about what is actually checked in CI during the review that I don't trust this way of testing anymore. But we could make a separate issue to track this if you'd rather prefer to merge what we have first.

Reply:
Ah, you're right, my description was wrong, but the test I'm doing still stands, I think (with the same reasoning but swapped file names).

I agree that this kind of testing is a bit brittle. I don't have any immediate improvements to suggest though, so I'd rather merge this and leave a better testing methodology for later.

Reply:
> but the test I'm doing still stands, I think (with the same reasoning but swapped file names).

I'm afraid this is not the case. You can try removing the autodiscovered DepOnExplicit.hs file and building to see that the test doesn't fail.

 I created https://github.com/tweag/gazelle_haskell_modules/issues/21